### ISSUE TITLE
apollo service:download doesn't work if OUTPUT doesn't exist.

### ISSUE THREAD

**Issue by taylorgoolsby** [apollographql/apollo-tooling#792]
I was originally using `graphql-cli` `graphql get-schema`. But `graphql-cli` said I should start using `apollo` instead.

However, the equivalent function `apollo service:download OUTPUT` is not able to download a schema if OUTPUT does not exist.

This is the error I get when I run it: `Error: Unable to read file ./schema.json. ENOENT: no such file or directory, open './schema.json'`.

Why is `service:download` looking for an existing OUTPUT? Isn't it suppoed to download and create the OUTPUT with the assumption that it doesn't exist yet?

If it's a feature and not a bug, then how do obtain `schema.json` in the first place to make `service:download` work? 
Why is `graphql get-schema` able to do this but `apollo service:download` isn't?

Here is my `apollo.config.js`:
```
module.exports = {
  service: {
    name: 'graphql',
    endpoint: {
      url: "http://localhost:4000/graphql"
    },
    localSchemaFile: "./schema.json"
  }
}
```

**Comment by inakianduaga** [comment]
For me, it's not working at all, getting 

```sh
✖ Loading Apollo Project
    → Cannot read property 'whenReady' of undefined
```

It's working correctly in version 1.9.2, so something must have broken in the latest release

**Comment by rfbotto** [comment]
@taylorgoolsby Had the same issue and setting the `url` and removing `localSchemaFile` on the `apollo.config.ts` seems to have solved the problem. The config seemed to have been looking for the `localSchemaFile` even when it was not generated hence the `Error: Unable to read file ./schema.json` 

```js
module.exports = {
  service: {
    name: 'graphql',
    url: "http://localhost:4000/graphql"
  }
}
```

**Comment by TLadd** [comment]
Ran into another variety of this when running `apollo service:download` where I had an outdated version of the schema. It appears that apollo performs a validation of the client queries before trying to download the new schema. Since my local copy of the schema was behind what my ui was expecting, there were types used in queries not defined in my copy of the schema, and there seemed to be no way to update it.

The trouble is that [Service Download](https://github.com/apollographql/apollo-tooling/blob/master/packages/apollo/src/commands/service/download.ts#L7) inherits from [Project Command](https://github.com/apollographql/apollo-tooling/blob/master/packages/apollo/src/Command.ts#L64), which always tries to [initialize a project](https://github.com/apollographql/apollo-tooling/blob/master/packages/apollo/src/Command.ts#L203) before doing anything else. When creating the project, it is expecting to be able to [load the schema](https://github.com/apollographql/apollo-tooling/blob/master/packages/apollo-language-server/src/project/client.ts#L113) and [validates](https://github.com/apollographql/apollo-tooling/blob/master/packages/apollo-language-server/src/project/base.ts#L111) that the client currently conforms to that schema it is able to load.

It seems to me that the `apollo service:download` ought to be exempt from this process and just give you the schema without trying to do validation beforehand, perhaps running validation of the client using the new schema after downloading it.

I was able to hack around this behavior by removing my apollog.config.js briefly:

```
rm apollo.config.js
apollo schema:download --endpoint=http://localhost:3001/graphql schema.json
git checkout apollo.config.js
```

**Comment by JakeDawkins** [comment]
I believe this has been fixed recently. Let me know if anyone can confirm it hasn't been.

**Comment by leonaves** [comment]
Not fixed in the latest released version (`2.16.0`).

**Comment by JakeDawkins** [comment]
I could be wrong, but I believe these issues are because of improper configuration, which is understandable, because the configuration for these projects isn't where it should be (which I'm actively working on).

The original issue is because the `service` config defined a `localSchemaFile` AND a `url`. Both of these are meant to be schema SOURCES, or where to look to find a schema. If the CLI is looking for a file that doesn't exist to build a schema from, it will error. That's why the second comment said their issue was fixed with removing the `localSchemaFile` option.

The other issue seen, mentioning trying to verify client operations against engine before downloading a service's schema is likely because they were trying to configure two projects–one client and one service in the same file. While we DO have that documented [on this guide](https://www.apollographql.com/docs/references/apollo-config/) as an option, I don't think the CLI **FULLY** supports this use case with **ALL** commands, and the outcome is unpredictable. I've opened a PR to clarify that and the purpose of the apollo config [here](https://github.com/apollographql/apollo/pull/533).

---

Id be happy to look into this further @leonaves if you can provide me a reproduction, but until then, I'm going to keep this closed, as I think it's configuration related.
